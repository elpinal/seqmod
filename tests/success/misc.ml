{
  signature S = {type t}
  type p = pack S
  module X = {
    val x = pack {type t = unit} : S
  }
  module Y = unpack X.x : S

  module F = fn X : {} => {}

  module M = {
    val id = fn x => x
  }

  module A = M :> {val id 'a : 'a -> 'a}
  module B = M :> {val id : unit -> unit}
}
