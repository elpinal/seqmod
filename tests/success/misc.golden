purity: impure
∃!50, !51, !52, !53, !54.{A:{id(=!53):∀!55.!55 -> !55}, B:{id(=!54):unit -> unit}, F:{} ->P {}, M:{id(=!52):∀!56.!56 -> !56}, X:{x(=!50):pack (∃!57.{t(=!57):Type})}, Y:{t(=!51):Type}, S=∃!58.{t(=!58):Type}, p(=pack (∃!59.{t(=!59):Type})):Type}
